compiler/transf.nim

```diff
@@ -734,7 +734,7 @@ proc transformFor(c: PTransf, n: PNode): PNode =
       stmtList.add(newAsgnStmt(c, nkFastAsgn, temp, arg))
       idNodeTablePut(newC.mapping, formal, temp)
 
-  let body = transformBody(c.graph, c.idgen, iter, true)
+  let body = transformBody(c.graph, c.idgen, iter, true).copyTree
```

Comment:
Seem expensive and worse, weird. How come for example that not a fresh set of local variables is required?

Reply:
It is a regression: https://github.com/nim-lang/Nim/commit/b1ff37c08f80b440e5ca0e4281a8967396efe7a5

Reply:
Before 0.20.0, copyTree is used.Then the optimization may remove it wrongly. Maybe I can make copyTree only apply to inline iterator with lifted variable s.